### DISCUSSION TITLE
RRF Support

### DISCUSSION THREAD

**Discussion by sb53systems** [sb53systems/G-Code-Flow-Temperature-Controller#13] [Ideas]
This discussion is created in order to help support RepRap firmware in this or similar script.

**Comment by Rickthebig** [comment]
Continuing the discussion about how to get this postprocessor to work with RRF gcode files, I am providing a new gcode file where all speeds are set to 200 mm/s, the temp. is set to 260 as a max. for PETG, and the volumetric flow limit is set to 13 mm3/sec - crazy settings, I hope your postprocessor does something reasonable out of it :-)
[3DBenchy_PETG.zip](https://github.com/user-attachments/files/16943405/3DBenchy_PETG.zip)

I would retry a run if I was able to get a Klipper style config.json file (or you direct me to the necessary changes in order to satisfy the expectations of your program). Why does your program not allow just to enter these values in the GUI?

**Reply by Rickthebig** [reply]
I found that the config.json I provided is completely irrelevant to the current configuration of my 3d printer. As I said, it was provided just to cheat your program; its content should not be regarded as valid.

Now I have been "reverse engineering" my environment to allow this postprocessor to run (without success), but at some point it will not make sense. What is really needed is a more generic version that focuses on the valid gcode commands, disregarding proprietary or special features or idiosyncracies of any kind of firmware as much as possible.

What are the specific obstacles in a RRF gcode file that prevents the postprocessor from running over it?

**Reply by sb53systems** [reply]
> Continuing the discussion about how to get this postprocessor to work with RRF gcode files, I am providing a new gcode file where all speeds are set to 200 mm/s, the temp. is set to 260 as a max. for PETG, and the volumetric flow limit is set to 13 mm3/sec - crazy settings, I hope your postprocessor does something reasonable out of it :-) [3DBenchy_PETG.zip](https://github.com/user-attachments/files/16943405/3DBenchy_PETG.zip)
> 
> I would retry a run if I was able to get a Klipper style config.json file (or you direct me to the necessary changes in order to satisfy the expectations of your program). Why does your program not allow just to enter these values in the GUI?

After editing your GCode file (delete 2 lines), the script was able to generate an almost good result  
  
![image](https://github.com/user-attachments/assets/e88c0dcb-835e-4c3d-afbe-a8be07c67787)  
![image](https://github.com/user-attachments/assets/ef78ae77-4773-4707-878b-8466b8554155)  
  
Note that :   
- this GCode file contains inconsistent data or slicer errors!  Orca Slicer and Klipper Estimator indicate that a layer height is missing in some layers, or a preview bug!  
![image](https://github.com/user-attachments/assets/89971e8e-c7c7-4157-948b-6085dcf19797)  
![image](https://github.com/user-attachments/assets/a762c4cd-f06f-428d-8e27-00689f25a3e2)  

- To estimate the time of each move in the GCode, Klipper Estimator needs the config.json file (must be obtained by the printer and not by the user) which contains the maximum speed and acceleration of your printer, as well as the information in the GCode such as the custom speeds and  acceleration for each line type.  
if we could find a solution that does the same with other Firmware, or try to reconcile Klipper's estimate with others by tinkering with the values and doing tests!

**Reply by sb53systems** [reply]
Another obstacle, the ability of the current version to detect the first movement to be taken into account in the GCode file.  
a future version could be based on the first line type indication, not the first GCode moves. I think this will allow us to work with any firmware by taking into account the variability of the parameter indications. Example bellow  
  
For Klipper  
![image](https://github.com/user-attachments/assets/c0cb4f78-db29-4f43-8c6e-fba474ac38e9)  
  
For Bambu  
![image](https://github.com/user-attachments/assets/81229b6a-e667-4aeb-af22-d963aba98c27)  
  
For RepRap  
![image](https://github.com/user-attachments/assets/abb1a3cc-ecf6-48b0-b984-0e3ba383f274)

**Reply by sb53systems** [reply]
@Rickthebig could you send a new GCode generated with Orca slicer ?  
your printer speed and acceleration limits,  
your recommanded Flow/Temperature,  
hotend temperature Rise/Fall time  
  
I'll try to generate a test GCode file that you could print

**Reply by Rickthebig** [reply]
Since now we can understand why/what is not adequate for the postprocessor, that is a good result I think.
- the issue with pre-print moves is manageable
- corrupted gcode? I will check wright now (looks healthy with the unchanged profile)
- config.json: The limits (speed, accel, jerk) are in the gcode file, due to this setting in Orca:
![image](https://github.com/user-attachments/assets/427c1973-9fd7-48bf-8f82-3e5609fc1033)
but the postprocessor does not look for them.

So config.json is needed for Klipper_Estimator, not directly for the postprocessor. Couldn't the postprocessor just read the estimated time directly from the gcode thus avoiding dependencies on external tools? Even if it is not as precise - not your fault. Orca will improve with time and generate better estimation times.

Let me explain, just in case: in RRF, the config.json is just a file to feed some starting settings for the online configuration tool. The Online Configuration Tool (https://configtool.reprapfirmware.org/) is a cool way to setup a new printer in RRF and it can read a config.json with some starting settings, but does not write one. Its output is a config.g and once the config.g is created, it is not in sync with the config.json anymore. Effectively, no config.json is needed to run a RRF printer. The assumption by the postprocessor that a valid config.json exists is a drawback and will only work for Klipper as it is (its format could even change or defer for other firmware flavors if present).

Currently, the only way to run the postprocessor on a RRF gcode file is to artificially setup a config.json file containing the correct limits - not a good solution on the long run. Getting rid of Klipper_Estimator and eliminating this dependency on config.json could lead to a more generic postprocessor, that could be used by any kind of firmware.

I wish I could engage in your programming project to help you out, but I am not proficient enough in this matter. So I hope that you finish these tiny bits of this marvelous postprocessor and change it to be of generic use (my advice: do it now while you are still deep in touch with the matter; if you postpone that work you will forget the details and it may never happen).

I intend to continue trying to run the postprocessor on my gcode files. First I will check what's up with the sliced benchy, 2nd I will painfully create a config.json by hand and 3rd I will remove those pre-printing lines prior to the postprocessing. I will come back and report my results.

Despite those difficulties let me complement you on such a remarkable quality improvement that this postprocessor apparently achieves. It will work at least for you :-)

**Reply by Rickthebig** [reply]
> could you send a new GCode generated with Orca slicer ?

I will do that later.
Thanks!

**Reply by sb53systems** [reply]
> Since now we can understand why/what is not adequate for the postprocessor, that is a good result I think.
> 
> * the issue with pre-print moves is manageable
> * corrupted gcode? I will check wright now (looks healthy with the unchanged profile)
> * config.json: The limits (speed, accel, jerk) are in the gcode file, due to this setting in Orca:
>   ![image](https://private-user-images.githubusercontent.com/151161212/366131383-427c1973-9fd7-48bf-8f82-3e5609fc1033.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5OTE1NjMsIm5iZiI6MTcyNTk5MTI2MywicGF0aCI6Ii8xNTExNjEyMTIvMzY2MTMxMzgzLTQyN2MxOTczLTlmZDctNDhiZi04ZjgyLTNlNTYwOWZjMTAzMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMFQxODAxMDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMzRlNTk0MDg4ZDU3ZjhlMjBkZjE0MTRlMzBjMTIyOGNmNWJkNzZkYzk2YTk4NzQzMWEzZjUxOTkzYzY3MjJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.a_TlLtg4CcCXZat0QXk-hZOUET6QdoMOHR28f32dTVU)
>   but the postprocessor does not look for them.
> 
> So config.json is needed for Klipper_Estimator, not directly for the postprocessor. Couldn't the postprocessor just read the estimated time directly from the gcode thus avoiding dependencies on external tools? Even if it is not as precise - not your fault. Orca will improve with time and generate better estimation times.
> 
> Let me explain, just in case: in RRF, the config.json is just a file to feed some starting settings for the online configuration tool. The Online Configuration Tool (https://configtool.reprapfirmware.org/) is a cool way to setup a new printer in RRF and it can read a config.json with some starting settings, but does not write one. Its output is a config.g and once the config.g is created, it is not in sync with the config.json anymore. Effectively, no config.json is needed to run a RRF printer. The assumption by the postprocessor that a valid config.json exists is a drawback and will only work for Klipper as it is (its format could even change or defer for other firmware flavors if present).
> 
> Currently, the only way to run the postprocessor on a RRF gcode file is to artificially setup a config.json file containing the correct limits - not a good solution on the long run. Getting rid of Klipper_Estimator and eliminating this dependency on config.json could lead to a more generic postprocessor, that could be used by any kind of firmware.
> 
> I wish I could engage in your programming project to help you out, but I am not proficient enough in this matter. So I hope that you finish these tiny bits of this marvelous postprocessor and change it to be of generic use (my advice: do it now while you are still deep in touch with the matter; if you postpone that work you will forget the details and it may never happen).
> 
> I intend to continue trying to run the postprocessor on my gcode files. First I will check what's up with the sliced benchy, 2nd I will painfully create a config.json by hand and 3rd I will remove those pre-printing lines prior to the postprocessing. I will come back and report my results.
> 
> Despite those difficulties let me complement you on such a remarkable quality improvement that this postprocessor apparently achieves. It will work at least for you :-)

Thank you for your help and advice.  
Be aware that we need a time estimate for each move to calculate the folw rate averages, and not of the overall printing time.  
  
Since I started this project and even if it's not supposed to be a definitive solution, I might as well try to make it more manageable, and give the possibility to other Firmware to try it!  
  
I will keep you informed.

**Reply by sb53systems** [reply]
Bellow the Klipper_Estimator Estimation/move exampl.  
![image](https://github.com/user-attachments/assets/51d14a9b-e7c4-475a-8a7a-e98c0d4fb00e)

**Reply by sb53systems** [reply]
Slicers work the same way and estimate time for other firmware less accurately.  
The integration of this concept into the Slicers will make it accessible to everyone with less effort and with more control and customization and more....  
  
My script is supposed to give the example and the possibility of testing under certain conditions, and could never follow the evolution of the software (Slicers) and the hardware (3D Printers).  
  
I would have liked to have found such a solution in the Slicer!

**Reply by Rickthebig** [reply]
> could you send a new GCode generated with Orca slicer ? your printer speed and acceleration limits, your recommanded Flow/Temperature, hotend temperature Rise/Fall time

I investigated the Benchy gcode I sent and it shows that there are two almost empty layers No. 53 and 54 (with small dots). This is not an anomaly, this is just how Orca works.
Take a look (that is the gcode reloaded back into Orca):
![image](https://github.com/user-attachments/assets/8e0b761a-aec7-4f4c-a9f1-79afc4b57852)
I also uploaded the gcode file to https://gcodeanalyser.com/ and it did not complain (there you can see many details and statistics). There are other online gcode analysers you can use to convince yourself.
What tool are you using that shows missing layers or any other slicing defects? Then I would look at it. Of course, you have to inspect the gcode *before* postprocessing it. Perhaps the missing layers are a consequence of postprocessing, a kind of bug or something, because your script is not expecting such a small layer (?).

I understand that time estimations per layer are needed, so dependency on an external tool remains.

These are my limits:
![image](https://github.com/user-attachments/assets/021e9e2a-0731-4ecf-971b-871d1c1f8c8a)
Those limits are at the begin of the executable part of the gcode the file (after the thumbnail lines), beginning with M201:
; EXECUTABLE_BLOCK_START
M486 S0 A"3DBenchy.stl_id_0_copy_0"
M486 S-1
M73 P0 R57
M201 X7000 Y7000 Z8000 E8000         ; <= sets max acceleration
M203 X19980 Y19980 Z1800 E7200     ; <= sets max feed rate
M204 P10000 T20000                           ; sets acceleration (P, T), mm/sec^2
M566 X3600.00 Y3600.00 Z1200.00 E600.00 ; sets the jerk limits, mm/min

The temperature of my nozzle changes quite fast, it reaches 245 Degrees C in less then 3 min. I estimate that it needs less than 10 sec to change 5 degrees and it cools down also quite rapidly as long as it is hot. There is no place in Orca to put that information though.

> I'll try to generate a test GCode file that you could print

If you generate a postprocessed gcode file (without layer defects) and send it to me then attach the corresponding config.json you used please. I will use it to do more experiments with different files (other models are more challenging and better suitable for showing the improvement of postprocessing).

But of course I need a postprocessed Benchy file without missing layers. Please investigate where those missing layers come from. For me, the gcode file is OK and if I slice a new one, it will be the same.

Do you know that there is a company (I think in Switzerland) that produces nozzles with the capability to extreme rapidly change temperature? Then one could set the temperature for every individual layer! The only problem with that nozzle is the price (about 450 $). But you see how forward-looking your postprocessor is.

Of course this ability is better implemented in the slicer but many such new features of slicers were first show-cased with a postprocessor, then the developers were convinced and integrated them :-)
Because Orca developers are overburdened with bug fixing, do not expect that they will take over this idea without compelling prove of concept.

Your turn.

**Reply by sb53systems** [reply]
@Rickthebig use discord ?!  
My discord is : sb53systems  
![image](https://github.com/user-attachments/assets/37cdc3a9-e3e4-41f0-adda-3cfad064324b)

**Reply by sb53systems** [reply]
Note that we need time for each move, not each layer  
  
I will soon finish a version that could recognize the beginning and end of printing, I will make an example of the config.json file, there you could generate and test a correct output file.  
  
i'll invistigate more and let you informed.

**Comment by sb53systems** [comment]
Here's the first file you could try:
[RRF_3DBenchy_PETG_Edited.zip](https://github.com/user-attachments/files/16959059/RRF_3DBenchy_PETG_Edited.zip)
  
Config.json file :  
[config.json](https://github.com/user-attachments/files/16959083/config.json)

![image](https://github.com/user-attachments/assets/e0efa833-b8c1-4b10-b891-8a6a40b3cdee)

**Reply by Rickthebig** [reply]
The Benchy file from [RRF_3DBenchy_PETG_Edited.zip](https://github.com/user-attachments/files/16959059/RRF_3DBenchy_PETG_Edited.zip) won't load into Orca:
![image](https://github.com/user-attachments/assets/2620506c-49db-495f-ae08-2266d1869402)
This is not a surprise, as it is corrupted with "214.7" inserted before every line line this:

214.7EADER_BLOCK_START    ; Reset Initial Temperature
214.7enerated by OrcaSlicer 2.1.1 on 2024-09-10 at 12:17:21    ; Reset Initial Temperature
214.7otal layer number: 240    ; Reset Initial Temperature
214.7ilament_density: 1.27    ; Reset Initial Temperature
214.7ilament_diameter: 1.75    ; Reset Initial Temperature
214.7ax_z_height: 48.00    ; Reset Initial Temperature
214.7EADER_BLOCK_END    ; Reset Initial Temperature
214.7    ; Reset Initial Temperature
214.7HUMBNAIL_BLOCK_START    ; Reset Initial Temperature
214.7    ; Reset Initial Temperature
214.7    ; Reset Initial Temperature
214.7humbnail begin 300x300 21232    ; Reset Initial Temperature

Something went wrong here.

Nonetheless I tried to process the original gcode file myself using the config.json you provided. I manually removed the two lines (Intro priming) in the gcode to facilitate recognizing the beginning of print operations, then loaded the gcode and printer config file and put the temp values in the GUI of the postprocessor. After clicking at "Generate" I get this:
![image](https://github.com/user-attachments/assets/6196dfea-17ad-48d5-ac97-47dd048522fa)
Unfortunately, I cannot edit the fields "Printer/Extruder Name" and "Filament Type" and these fields remain unchanged (empty) after the "Generate" operation. I don't know what your postprocessor is expecting to find in the gcode file to define Printer/Estruder Name and Filament Type.

What am I doing wrong?

**Reply by sb53systems** [reply]
sorry it was a bug that I hadn't noticed so much I was tired.  
Here is the part detected and modified by the script (without the initial G-Code instructions):  
[tempOut.zip](https://github.com/user-attachments/files/16966055/tempOut.zip)  
  
In a few hours I will have finished the restoration of the file, then I will upload it to GitHub so that you can try it.  
  
To create a profile on the script just press: Save and choose a name after (Like Orca)  
![image](https://github.com/user-attachments/assets/fa911499-b99a-4d59-959b-848ecaaa2daf)

**Reply by Rickthebig** [reply]
I found the hotend that is able to change temperature from room temperature to 200 deg.C in 5 seconds! Here it is: Plasmics INo Trident (Austrian company) Usual variations in temperature while printing can be achieved instantly. It started as a Kickstarter project but is now readily available to buyers.
https://plasmics.com/inotrident
It is mentioned in a CNC Kitchen video about Formnext here: https://youtu.be/8k89c2iSU60?t=320

If I was in your position, since you have some experience and even a suitable tool, I would contact the Plasmics company seeking for a cooperation - you would get that hotend for free I guess (I would first improve the postprocessor though).

Interestingly, later in the same video, it is mentioned that for a giant 3d printer using a variable nozzle size, a *postprocessor* is being used (https://youtu.be/8k89c2iSU60?t=442) - this is because no slicer supports variable nozzle size... You see, you are in trend.

**Reply by sb53systems** [reply]
I just uploaded an update to the \Bin folder  
download the script from there (SB53-Systems.exe).  
![image](https://github.com/user-attachments/assets/20e1c923-d26c-43cb-8b20-fbe06e0bc733)  
  
It gives a pretty decent result, I think!  
![image](https://github.com/user-attachments/assets/c2e372f1-88a4-4f33-8e29-16e325139b85)  
  
From this  
![image](https://github.com/user-attachments/assets/f6833c4f-9748-4fc4-9041-81a9bf1d165a)  
  
To This  
![image](https://github.com/user-attachments/assets/8e37aba0-4103-4f7d-a867-3d3778a9ccc9)

**Reply by sb53systems** [reply]
You just have to watch out for a last bug that adds a temperature change at the end caused by a custom command  
![image](https://github.com/user-attachments/assets/f8b5878b-dbbc-408f-9a8d-f6f9a18f0788)  
  
Let me know if....

**Reply by sb53systems** [reply]
@Rickthebig Given my level of English, I can't explain everything quickly (using a translator), I think Discord is more useful for screen sharing!

**Reply by Rickthebig** [reply]
I checked that line
;M109 S41        ; set Extruder T to 41 and wait - ;G10/M568 R50 S50 - followed by ...
in my end code. It is a comment (disabled). For some reason, it appears as a valid line in the postprocessed gcode file, don't know why. You perhaps? :-)  Take care about multiple comment signs in the same line!

**Reply by sb53systems** [reply]
> I checked that line ;M109 S41 ; set Extruder T to 41 and wait - ;G10/M568 R50 S50 - followed by ... in my end code. It is a comment (disabled). For some reason, it appears as a valid line in the postprocessed gcode file, don't know why. You perhaps? :-) Take care about multiple comment signs in the same line!
 
The original one in the left, The line was not commented.  
![image](https://github.com/user-attachments/assets/baeada71-e786-49cd-814e-a2668f6b878d)

**Reply by sb53systems** [reply]
I think it's the only detail that is missing (to be deleted manually)  
  
with the updated version you could try adjusting config.json values to your real printing time.  
Let me know if you have any issues with script profiles

**Reply by sb53systems** [reply]
The script was updated a short time ago (in \bin),

**Reply by sb53systems** [reply]
Here is your generated GCode  
[RRF_3DBenchy_PETG_Edited.zip](https://github.com/user-attachments/files/16969115/RRF_3DBenchy_PETG_Edited.zip)  
[config.json](https://github.com/user-attachments/files/16969141/config.json)  
  
![image](https://github.com/user-attachments/assets/71bf59dd-6e7f-41b1-970f-9bc5ac7dc50d)  
![image](https://github.com/user-attachments/assets/436d6646-a324-4fc7-a368-f65788fc4ef9)

**Reply by sb53systems** [reply]
I advise you not use Ironing, this will affect speed and quelity.  
  
I look forward to your feedback :-)

**Reply by Rickthebig** [reply]
Preliminary results and remarks after printing the postprocessed Benchy and also a non-postprocessed for comparison:

1. The temperature rises to the maximum value (set to 260 C) just when the gcode is started, but sinks suddenly to the minimum value (set to 210 C) before the 1st printing move - this is unnecessary.
2. The 1st layer is printed at 210 C for PETG. This is not appropriate and I think the 1st layer should be left unchanged from any postprocessing.
3. The hotend temperature was most of the time at about 220 C - for PETG on the lower end, but it can be changed of course. During printing, the temperature changed 2-3 degrees up and down on visually homogeneous parts. The hotend was slower than assumed, and at times, its actual temperature differed 5 C from the set temperature. The reaction time for temperature changes can also be adapted in the postprocessor - providing this possibility is a smart move from you.
4. The original post-printing macro has some settings to allow a controlled shut down after the temperature of the hotend reaches 45 C, but due to the postprocessing that did not work (you already noticed the bug). That should be addressed, because it means that the hotend and heating bed will indefinitely be kept at a high temperature after the print is over (this is an issue specially when printing overnight).
5. Comparing the two printed parts, the non-postprocessed looks better. There are no really severe quality glitches in the postprocessed part, but it shows stains and marks where the surface should be just smooth (as it is in the non-postprocessed part). They cannot be caused just by small changes in temperature because they are quite tangible and it looks like the flow was overmodulated. You wrote that the postprocessor not only limits the speed according to the set volumetric limit, it also adapts the flow corresponding to the modulated temperature - but how is this dependency defined? I cannot see any real benefit from the postprocessing in this particular case (I already mentioned that a Benchy is not suitable to showcase the benefits of temperature control) and as it is, there are a few things that need to be tackled.

Do you adapt the fan speed in any way?
I will upload some pictures later.

**Reply by sb53systems** [reply]
@Rickthebig , dont expect a good result with no time estimation compatibility and no calibrations!  
I think that with more tests, you could get a correct result.  
  
Note that :   
- You need to compare and confirm that the time estimated by Klipper_Estimator script is close to the your real printing time.  
- You need to adapt the recommended Flow/temperature to your hotend.  
- The first layer temperature is adapted to the speed, line width and layer Height of the first moves of the GCode.  
  
Send me your printer, filament and process presets .zip file, I will analyze your configurations and understand better.

**Reply by Rickthebig** [reply]
Now I am trying to use the postprocessor with a different model, which seems more suitable for this purpose, but I have no luck with this postprocessor.

I still cannot go past this error message:
![image](https://github.com/user-attachments/assets/1642442b-9118-4640-ad55-96651456c704)

The required information is already in the gcode file, at the end (as well as all other settings):
; filament_type = PETG
; printer_model = DIY BLV Cube

But the postprocessor insists that they are not found (perhaps is the format different for Klipper). What is an "Extruder" in this context? The error tells me to "choose my settings and refresh". Where should I choose my settings? (The gcode file is already loaded).

At this point, I am stuck.

I would provide the presets, but export fails (Orca is quite buggy) because whatever I choose I get an error:
![image](https://github.com/user-attachments/assets/79e0752c-4616-4bed-a40a-1a4be88f64d2)
https://github.com/SoftFever/OrcaSlicer/issues/4578

**Reply by sb53systems** [reply]
Even if the informations are not found in the GCode file, you could use the scrip preset that you want. choose a preset then Click in Refresh Estimation, then Click in Generate.  
  
Note :  
![366987402-1642442b-9118-4640-ad55-96651456c704](https://github.com/user-attachments/assets/1ca902ae-826a-4564-8873-cbe314349207)
  
You have to set the correct Flow/Temperature values, and delete the Macro and Extruder name. exampl bellow.  
I recommand you to use 30 in the Max/Average Smoothing value.  
  
![image](https://github.com/user-attachments/assets/c168a141-067e-46a8-897e-a60a917124aa)

**Reply by sb53systems** [reply]
Try exporting Orca presets separately and choose your current preset.  
  
![image](https://github.com/user-attachments/assets/74a0fade-5eae-47bd-9b70-ab20fdce5fea)

**Reply by sb53systems** [reply]
Try to follow the instructions for use provided on the homepage.  
  
the script can recognize the Extruder/Printer and Filament used if they are written with the same name as in the slicer.  
  ![image](https://github.com/user-attachments/assets/7b467275-4bed-4927-adc6-0a6306d95de6)![image](https://github.com/user-attachments/assets/0219a6d0-63d1-4b7e-b465-d45c74db0d49)  
  ![image](https://github.com/user-attachments/assets/40a1eeec-38eb-4db5-a4bf-36e83e9e284f)![image](https://github.com/user-attachments/assets/fefd7247-1c32-4f94-a1b5-1fc3124d0812)

**Reply by Rickthebig** [reply]
> Send me your printer, filament and process presets .zip file, I will analyze your configurations and understand better.

Since I cannot export them, I am providing a project file with a model and all settings customized for the postprocessor - no idea if you will be able to use that project file.
At the moment, I am not able to proceed with any tests (I could never get the postprocessor to convert any gcode file).

[AIl_In_One_test_nano-TempContr-PETG-ORCA.zip](https://github.com/user-attachments/files/16983714/AIl_In_One_test_nano-TempContr-PETG-ORCA.zip)

**Reply by sb53systems** [reply]
@Rickthebig Is there a way to exchange on another platform (like Discord)?

**Reply by Rickthebig** [reply]
![IMG_20240912_192404](https://github.com/user-attachments/assets/7d33a3f3-d665-4f6a-a17a-7d34ba280a2a)
![IMG_20240912_192514](https://github.com/user-attachments/assets/8538b873-5bfe-41aa-a17a-52a39bcbea73)
On the left side the original part, on the right side the postprocessed one (ignore the different color, it is a rainbow PETG filament).
In order to show the defects caused by the postprocessor, look at this:
![IMG_20240912_192819](https://github.com/user-attachments/assets/24f81637-778f-4148-9788-9f3ba6dc4ef0)
The defects are not just optical, but it looks worse as it is.

**Reply by sb53systems** [reply]
@Rickthebig Send me another 3DBenchy GCode file without Ironing,

**Reply by sb53systems** [reply]
Exampl :  
![image](https://github.com/user-attachments/assets/80e84f54-3d32-41cc-8263-fc94d93450b5)

**Reply by Rickthebig** [reply]
I am getting a bit closer. After naming the printing and filament exactly as they were used to create the gcode file
![image](https://github.com/user-attachments/assets/ca5fa53d-eba1-480e-83d6-6ef6056d9da5)
I retried and got this:
![image](https://github.com/user-attachments/assets/ea47ce76-18f7-4e67-9dd4-75a0ef10a72d)
Please note that the flow settings were reset (originally, I entered 1, 8 and 15 mm3/s for each temperature respectively). Apparently the input gcode was analysed but not output generated. The error looks like a program crash.

This was my input gcode file:
[Uploading All_In_One_3D_printer_test_nano_PETG.zip…]()

**Reply by sb53systems** [reply]
@Rickthebig you are using an old version of the script, download the last updated version

**Reply by sb53systems** [reply]
> Please note that the flow settings were reset (originally, I entered 1, 8 and 15 mm3/s for each temperature respectively).   
  
you have to save your new values

**Reply by Rickthebig** [reply]
> Try to follow the instructions for use provided on the homepage.

If the postprocessor does not run when the presets were not exported, then I have no chance to generated a converted gcode file. I have used the same preset names as in the slicer hoping that so it could work. In my case, the export of presets in Orca is *broken* (this is a documented bug). Are you using a nightly build or the regular version 2.1.1? It could be that this bug is already fixed in the nightly builds.

If I could use this postprocessor at last, I would be able to do lots of tests. The test with the Benchies was just a preliminary run, and as you said, the flow at different temperatures must be adjusted. This is an interesting point - how to calibrate that?

**Reply by sb53systems** [reply]
![image](https://github.com/user-attachments/assets/e4bba5ba-0dc1-45e3-af24-e4db07e884f4)
  
To calibrate the edeal flow by temperature, I do a visual test of a vase mode print with a fixed flow rate in the Slicer, and then manually change the temperature during printing.  (later I'll add more presision)

**Reply by Rickthebig** [reply]
At last, using the latest version:
![image](https://github.com/user-attachments/assets/8bdf7020-e305-4325-8762-ff355a4c5ab4)

Do you have a model demonstrating an improvement after postprocessing?

And yes, we can switch to Discord. Your good English will be better then :-)

**Reply by Rickthebig** [reply]
In Discord, there is nothing I can do:
![image](https://github.com/user-attachments/assets/5ed9e973-c8d5-4361-a4bc-afd59fb73633)

**Reply by sb53systems** [reply]
> And yes, we can switch to Discord. Your good English will be better then :-)  
  
My Discord : sb53systems

**Reply by sb53systems** [reply]
send you discord psudo to my email

**Reply by sb53systems** [reply]
Can you post a reply on the orca discord server?  
![image](https://github.com/user-attachments/assets/2fb6e61a-2ca9-4bf0-8977-1b360c0c3d1a)

**Reply by Rickthebig** [reply]
My Discord name is "TrueCubed". 
I am currently logged in in the OrcaSlicer server.
Is there a SB53Systems server? I have no idea how to reach you other than in the OrcaSlicer Server.

I am now printing the postprocessed file...

**Reply by sb53systems** [reply]
@Rickthebig , i snet you a message in discord
